pkg/didcomm/dispatcher/api_test.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package dispatcher
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+const errTimeout = "timeout: the result was not received"
+
+func TestNewDIDCommCallback(t *testing.T) {
+	require.NotNil(t, NewDIDCommCallback())
+}
+
+func TestDIDCommCallback_Done(t *testing.T) {
+	t.Run("WithoutResult", func(t *testing.T) {
+		c := NewDIDCommCallback()
+
+		err := c.Done(Result{})
+		require.NoError(t, err)
+		require.True(t, c.done)
+
+		const errMsg = "callback was already done use IsValid() function to prevent this error"
+		// double done
+		err = c.Done(Result{})
+		require.EqualError(t, err, errMsg)
+	})
+
+	t.Run("WithResult", func(t *testing.T) {
+		c := NewDIDCommCallback().WithResult()
+		done := make(chan bool)
```

Comment:
See https://github.com/hyperledger/aries-framework-go/pull/321#discussion_r327313409

FYI: done channel signatures are usually this:
`chan struct{}`